examples/seismic/model.py

```diff
@@ -386,6 +389,18 @@ def __init__(self, origin, spacing, shape, space_order, vp, nbpml=20,
         else:
             self.phi = 0
 
+    def physical_params(self, m, **kwargs):
+        """
+        Return all set physical parameters
+        """
+        all_phys = [self.m, self.epsilon, self.delta, self.theta, self.phi]
```

Comment:
Here's what I'd rather do. 

You could define the _names_ of the functions representing physical parameters at the class level. Ie,
```
class Model(object):
    _physical_parameters = ('m', 'epsilon', ...)
```

Finally, you can change the implementation as:
```
def defined_parameters(self, **kwargs):
    known = [getattr(self, i) for i in Model._physical_parameters]
    return {i.name: kwargs.get(i.name, i) for i in known if i is not None}
```

However, I'm struggling to understand why we need this contrived logic...

Reply:
We need this because in practice, we dont alwyas have all the TTI parameters and current setup forces all or nothing.

Reply:
> and current setup forces all or nothing.
so that's what we should fix

Reply:
And this what does. This merly check what's in the model and what should be ignore for the operator call.